circuit_maintenance_parser/parser.py

```diff
@@ -175,7 +179,23 @@ class Html(MaintenanceNotification):
     _data_type = "text/html"
 
     def process(self) -> Iterable[Maintenance]:
-        """Method that returns a list of Maintenance objects."""
+        """Execute parsing."""
+        data: Dict[str, Union[int, str, Iterable]] = {
+            "provider": self._default_provider,
+            "organizer": self._default_organizer,
+        }
+        try:
+            soup = bs4.BeautifulSoup(quopri.decodestring(self.raw), features="lxml")
+
+            self.parse_html(soup, data)
+
+            return [Maintenance(**data)]
```

Comment:
I like moving the `quopri.decodestring()` to the base class, but I wonder if constructing the `[Maintenance(**data)]` in the base class is overly limiting - do we anticipate any case where a single `self.raw` would instead result in more than one `Maintenance` object being constructed?

Reply:
I wanted to not limit a notification to only one maintenance even all the notifications we have seen so far contained only one maintenance, but being able to get ready for the more generic use case that contains more than one maintenance.
But maybe you are right and we could simplify it to one `raw` to one `Maintenance`, and now that we are in early stage of the project is a good time to do it...

Reply:
I actually meant things the other way round - if we do want to permit the case where more than one maintenance are returned, the code you have here as written is overly restrictive; I would want to let `parse_html` return a list of (one or more) `Maintenance` objects itself rather than hard-coding things to always construct exactly one such object.

Reply:
gotcha.
I started first with iCalendar, that supports multiple EVENTS in one notifications, so I hardcoded the same signature behaviour for HTML even I have never seen any one with this behaviour, so I ended up with always returning a list of 1 element, as you noticed.
To make this code ready to accommodate this use-case, I have changed a bit the method to handle future HTML parsers that could contain multiple maintenances.